runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/Bindings.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc.
+ *
+ * Aklivity licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.aklivity.zilla.runtime.engine.internal;
+
+import static java.nio.ByteOrder.nativeOrder;
+import static java.nio.file.StandardOpenOption.WRITE;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.agrona.LangUtil;
+
+import io.aklivity.zilla.runtime.engine.config.BindingConfig;
+
+public final class Bindings implements AutoCloseable
+{
+    private static final int RECORD_SIZE = 5 * Long.BYTES;
+
+    private final Path path;
+    private final ByteChannel channel;
+    private final Map<Long, long[]> bindings;
+
+    private Bindings(
+        Path path,
+        ByteChannel channel,
+        Map<Long, long[]> bindings)
+    {
+        this.path = path;
+        this.channel = channel;
+        this.bindings = bindings;
+    }
+
+    public void writeBindingInfo(
+        BindingConfig binding)
+    {
+        ByteBuffer byteBuf = ByteBuffer.wrap(new byte[RECORD_SIZE]).order(nativeOrder());
+        byteBuf.putLong(binding.id);
+        byteBuf.putLong(binding.typeId);
+        byteBuf.putLong(binding.kindId);
+        byteBuf.putLong(binding.originTypeId);
+        byteBuf.putLong(binding.routedTypeId);
+        byteBuf.flip();
+        while (byteBuf.hasRemaining())
+        {
+            try
+            {
+                channel.write(byteBuf);
+            }
+            catch (IOException ex)
+            {
+                System.out.printf("Error: %s is not writeable\n", path);
```

Comment:
Is the exception message not already being printed out anyway by ZillaStartCommand?

Same feedback for other occurrences below.